src/Cryptography/Sodium/Bindings/SHA512.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE Trustworthy #-}
+
+-- |
+--
+-- Module: Cryptography.Sodium.Bindings.SHA512
+-- Description: Direct bindings to the SHA-512 hashing function implemented in Libsodium
+-- Copyright: (C) Hécate Moonlight 2022
+-- License: BSD-3-Clause
+-- Maintainer: The Haskell Cryptography Group
+-- Stability: Stable
+-- Portability: GHC only
+module Cryptography.Sodium.Bindings.SHA512
+  ( -- * Introduction
+    -- $introduction
+
+    -- * Single-part message
+    cryptoHashSHA512,
+
+    -- * Multi-part messages
+    CryptoHashSHA512State (..),
+    cryptoHashSHA512Init,
+    cryptoHashSHA512Update,
+    cryptoHashSHA512Final,
+    -- * Constants
+    cryptoHashSHA512Bytes,
+  )
+where
+
+import Foreign (Ptr)
+import Foreign.C (CInt (..), CSize (..), CUChar (..), CULLong (..))
+
+-- $introduction
+--
+-- The SHA-512 functions are provided for interoperability with other applications. If you are
+-- looking for a generic hash function and not specifically SHA-2, using
+-- 'Cryptography.Sodium.Bindings.GenericHashing' (BLAKE2b) might be a better choice.
+-- These functions are also not suitable for hashing passwords or deriving keys from passwords.
+-- Use 'Cryptography.Sodium.Bindings.PasswordHashing' instead.
```

Comment:
Would be better phrased as "If this is what you need, use 'Cryptography.Sodium.Bindings.PasswordHashing' instead.".